src/OrleansCodeGenerator/RoslynCodeGenerator.cs

```diff
@@ -309,12 +312,21 @@ private static GeneratedSyntax GenerateForAssemblies(List<Assembly> assemblies,
                 var assembly = assemblies[i];
                 foreach (var attribute in assembly.GetCustomAttributes<ConsiderForCodeGenerationAttribute>())
                 {
-                    ConsiderType(attribute.Type, runtime, targetAssembly, includedTypes, attribute.GenerateSerializer);
+                    ConsiderType(attribute.Type, runtime, targetAssembly, includedTypes, true);
+                    if (attribute.ThrowOnFailure && !SerializerGenerationManager.IsTypeRecorded(attribute.Type))
+                    {
+                        throw new CodeGenerationException(
+                            $"Found {attribute.GetType().Name} for type {attribute.Type.GetParseableName()}, but code"
+                            + " could not be generated. Ensure that the type is accessible.");
+                    }
                 }
 
+                KnownAssemblyAttribute knownAssemblyAttribute;
+                var treatAsSerializable = knownAssemblyAttributes.TryGetValue(assembly, out knownAssemblyAttribute)
```

Comment:
that would probably be cleaner, yes. Nevertheless we'd like to kick off a build of 1.2.3 very shortly today, so we'll merge this in as is, and we can clean up the code later if you want (it's not a big deal anyway).
Unless you tell me you can do this change asap, I'll go ahead an merge in half an hour.